### DISCUSSION TITLE
Change mouse cursor

### DISCUSSION THREAD

**Discussion by BladeTransformerLLC** [emilk/egui#4988] [Q&A]
Hi there!

I'd like to change egui's mouse cursor depending on the state of the app.

I have a renderer loop that looks like:

```
gui.update(
	&mut frame_input.events,
	frame_input.accumulated_time,
	frame_input.viewport,
	frame_input.device_pixel_ratio,
	|gui_context| {
		//gui_context.output().cursor_icon = egui::CursorIcon::Wait; // doesn't work
		//gui_context.set_cursor_icon(egui::CursorIcon::Wait);  // doesn't work
		//gui_context.output_mut(|o| o.cursor_icon = egui::CursorIcon::Wait);  // doesn't work
		
		//...
	},
);
```

What is the proper way to change it? [This answer](https://github.com/emilk/egui/discussions/858) doesn't seem to work anymore.

Thanks

**Comment by antoniocorbi** [comment]
Hi,

I've got the same problem (egui 0.31.1). Is there any update on this?
Thx!

**Comment by nnmarcoo** [comment]
`gui_context.set_cursor_icon(CursorIcon::Wait);` works for me. You may be inadvertently changing the icon somewhere else in your code?

**Reply by antoniocorbi** [reply]
> `gui_context.set_cursor_icon(CursorIcon::Wait);` works for me. You may be inadvertently changing the icon somewhere else in your code?

I've opened an issue about this. 

A the time of the problem I was using gnome-wayland as my desktop. I decided to try sway and exactly the same code that does not work (for me) in gnome works ok in sway.
